src/component.cpp

```diff
@@ -136,5 +138,72 @@ void Component::addComponent(const Component &c)
     mPimpl->mComponents.push_back(c);
 }
 
+void Component::removeComponent(const std::string& name)
+{
+    auto result = mPimpl->findComponent(name);
+    if(result != mPimpl->mComponents.end()) {
+        mPimpl->mComponents.erase(result);
+    } else {
+        throw std::out_of_range("Named component not found.");
```

Comment:
It would be handy if these exceptions strings had the function name in them... e.g. __func__ or __FUNCTION__, as long as its portable.